src/App.tsx

```diff
@@ -17,18 +79,31 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                getFilter={getFilter}
+                selectedFilter={selectedFilter}
+                searchBarValue={searchBarValue}
+                getSearchBarValue={getSearchBarValue}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {!todos.length && <Loader />}
+              <TodoList
+                todos={filteredTodos}
+                selectedTodoId={selectedTodo?.id}
+                showTodo={showTodo}
+              />
             </div>
           </div>
         </div>
       </div>
-
-      <TodoModal />
+      {selectedTodo && (
+        <TodoModal
+          closeModal={closeModal}
+          currentTodo={getCurrentTodo()}
```

Comment:
do you really need this function?
suggest pass currentTodo.
and change type of currentTodo prop in TodoModal